frontend/src/pages/projects/screens/spawner/storage/usePreferredStorageClass.ts

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import { AppContext } from '~/app/AppContext';
+import { StorageClassKind } from '~/k8sTypes';
+
+const usePreferredStorageClass = (): StorageClassKind | undefined => {
+  const {
+    dashboardConfig: {
+      spec: { notebookController },
+    },
+    storageClasses,
+  } = React.useContext(AppContext);
+
+  const defaultClusterStorageClasses = storageClasses.filter((storageclass) =>
+    storageclass.metadata.annotations?.['storageclass.kubernetes.io/is-default-class']?.includes(
+      'true',
+    ),
+  );
+
+  const configStorageClassName = notebookController?.storageClassName ?? '';
+
+  if (defaultClusterStorageClasses.length != 0) {
+    return undefined;
```

Comment:
Small nit... the difference in semantics of `undefined` and `null` are usually never cared about by many. But `undefined` means "no answer" and `null` means "nothing". Technically this and all the other returns should be `null`.

Reply:
I am not sure how later on, the assemblePVC would react to storageClassName: null, only know the behavior with undefined.

Reply:
`?.` does both null and undefined -- it's not a big deal. we can leave this too.

Reply:
yes, code-level in typescript, no problem, but later on, when the pvc json is assembled and passed with @openshift/dynamic-plugin-sdk-utils, 

https://github.com/opendatahub-io/odh-dashboard/blob/20162439f1ffd3e59423de9fa54ffe649a656da8/frontend/src/api/k8s/pvcs.ts#L40

I don't know how it would handle it. So I prefer keeping undefined here, as that worked without a doubt up end2end on the cluster.